closed/make/Docs.gmk

```diff
@@ -20,4 +20,77 @@
+
+################################################################################
+# The javadoc tool only allows for a single setting for headers and footers
+# per invocation. The sed command below replaces the footer for the pages
+# relating to openj9 classes to the correct value for those classes.
+#
+# '.' need escaping in the sed regex. If more special characters are introduced
+# in the openjdk JAVADOC_BOTTOM in future these would need escaping also.
+#
+# '&' needs escaping in sed replacement text.
+#
+# {@docroot} is expanded by javadoc when the html files are created, so the matching
+# needs to cater for anything {@docroot} may have been substituted by - '../../../..'
+# or '../..' or '..' etc.
+
+# \(\.\./\)*\.\. means zero  or more '../' strings followed by '..'
+
+JAVADOC_DOCROOT=\(\.\./\)*\.\.
+JAVADOC_BOTTOM_REGEX = $(subst {@docroot},$(JAVADOC_DOCROOT),$(subst .,\.,$(JAVADOC_BOTTOM)))
+OPENJ9_JAVADOC_BOTTOM := $(subst &,\&,$(OPENJ9_JAVADOC_BOTTOM))
+OPENJ9_SUBST_COMMAND = sed -i 's|$(JAVADOC_BOTTOM_REGEX)|$(OPENJ9_JAVADOC_BOTTOM)|g'
+
+##################################################################################
+# Define a javadoc Group for the OpenJ9 specific modules
+# Module names are added to DOCS_MODULES in closed/make/common/Modules.gmk
+OPENJ9_GROUP_NAME := OpenJ9
+OPENJ9_GROUP_MODULES := openj9.*
+OPENJ9_GROUP_DESCRIPTION := \
+    Modules whose names start with {@code openj9.} contain APIs provided by the \
+    <a href="$(OPENJ9_JAVADOC_BASE_URL)">Eclipse OpenJ9</a> project. \
+    These APIs can only be used with a JDK which includes the OpenJ9 virtual machine. \
+    #
+JDK_GROUPS += OPENJ9
+
+##################################################################################
+# Add openj9 specific modules to the list of modules for which javadoc is required
+OPENJ9_MODULES := $(filter openj9.%,$(call FindAllModules))
+DOCS_MODULES += $(OPENJ9_MODULES)
+
+OPENJ9_JAVADOC_MARKER := $(SUPPORT_OUTPUTDIR)/docs/_javadoc_JDK_API_openj9.marker
+JDK_API_CUSTOM_TARGETS += $(OPENJ9_JAVADOC_MARKER)
+
+##################################################################################
+# Replace the html footer in the generated javadoc for openj9 specific files.
+# Using the OPENJ9_JAVADOC_MARKER file as a target ensures the page footers are
+# only updated if the javadoc has been regenerated.
+
+$(OPENJ9_JAVADOC_MARKER): $(DOCS_OUTPUTDIR)/api/index.html
+	# Replace footer in the pages for
+	# 1. openj9 modules, packages and classes
+	# 2. com.ibm packages and classes
+	# 3. the com.sun.management classes reimplemented in OpenJ9
+	$(CD) $(DOCS_OUTPUTDIR)/api; \
+	find . -type f \
+        \( -path "*/openj9*/*.html" \
+        -o -path "*/com/ibm/*/*.html" \
+        -o -name GarbageCollectionNotificationInfo.html \
+        -o -name GcInfo.html \
+        -o -name HotSpotDiagnosticMXBean.html \) \
+    | xargs -I {} $(OPENJ9_SUBST_COMMAND) {}
```

Comment:
Please indent these lines to clarify that they are continuations of a single command.

Reply:
Reformatted